connector/protobuf/src/test/scala/org/apache/spark/sql/protobuf/ProtobufCatalystDataConversionSuite.scala

```diff
@@ -123,16 +123,21 @@ class ProtobufCatalystDataConversionSuite
     StringType -> ("StringMsg", ""))
 
   testingTypes.foreach { dt =>
-    val seed = 1 + scala.util.Random.nextInt((1024 - 1) + 1)
+    val seed = scala.util.Random.nextInt(RandomDataGenerator.MAX_STR_LEN)
     test(s"single $dt with seed $seed") {
 
       val (messageName, defaultValue) = catalystTypesToProtoMessages(dt.fields(0).dataType)
 
       val rand = new scala.util.Random(seed)
       val generator = RandomDataGenerator.forType(dt, rand = rand).get
       var data = generator()
-      while (data.asInstanceOf[Row].get(0) == defaultValue) // Do not use default values, since
-        data = generator()                                  // from_protobuf() returns null in v3.
+      // Do not use default values, since from_protobuf() returns null in v3.
+      while (
+        data != null &&
```

Comment:
This should just be `data == null || data.asInstanceOf[Row].get(0) == defaultValue`?

Reply:
I meant, we don't need the array check.

Reply:
@rangadi 
1. `data.asInstanceOf[Row].get(0) == ByteString.empty().toByteArray`
2. `data.asInstanceOf[Row].get(0) == Array.emptyByteArray`
3. `data.asInstanceOf[Row].get(0) ==  ByteString.EMPTY.toByteArray`
4. `data.asInstanceOf[Row].get(0) ==   "".getBytes`
5. `data.asInstanceOf[Row].get(0).isInstanceOf[Array[Byte]] && data.asInstanceOf[Row].get(0).asInstanceOf[Array[Byte]].isEmpty`

Except for (5), none of them worked. I'm printing under the conditions listed above.